src/components/Header/Header.test.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { render, screen, cleanup } from '@testing-library/react';
+
+import App from '../App/App.component';
+
+describe('App Component Tests', () => {
+  const setState = jest.fn();
+  const useStateSpy = jest.spyOn(React, 'useState');
+  useStateSpy.mockImplementation((init) => [init, setState]);
+
+  beforeEach(() => {
+    render(<App />);
```

Comment:
Here you should only render the `<Header/>` component to do unit testing on it, if you render the App component, you are renderin the whole app to this test suit

Reply:
Changed